### PR TITLE
Remove nestingLevel JS code

### PR DESCRIPTION
**Pull request by ntrel**
We can use `DDOC_ANCHOR`'s argument instead, which is fully qualified.

### CODE REVIEW

File: std.ddoc [hunk 1]
```diff
@@ -37,13 +44,14 @@ function listanchors()
     values.sort();
 
     var newText = "";
-    for(var i = 0; i < values.length; i++) {
+    for (var i in values) {
```
Comment (MartinNowak) [inline std.ddoc]: Rather keep the `for(var i = 0; i < values.length; ++i)` loop.

File: std.ddoc [hunk 2]
```diff
@@ -15,16 +15,23 @@ DDOC = <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
...
 //
 function defineSymbol(sym) {
-  if (nestingLevel > 0) return;
+  // ignore ditto overloads
   if (sym == lastSymbolDefined) return;
-  if (true) lastSymbolDefined = sym;
+  // ignore scoped symbols, but show module-level overloads
+  var pos = sym.indexOf('.');
+  if (pos >= 0 && (sym.lastIndexOf('.') != pos || isNaN(lastName(sym)))) return;
+  // Note: empty comment avoids macro def
```
Comment (MartinNowak) [inline std.ddoc]: What does that mean?

File: std.ddoc [hunk 3]
```diff
@@ -15,16 +15,23 @@ DDOC = <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
...
+}
 //
 function defineSymbol(sym) {
-  if (nestingLevel > 0) return;
+  // ignore ditto overloads
   if (sym == lastSymbolDefined) return;
-  if (true) lastSymbolDefined = sym;
+  // ignore scoped symbols, but show module-level overloads
+  var pos = sym.indexOf('.');
+  if (pos >= 0 && (sym.lastIndexOf('.') != pos || isNaN(lastName(sym)))) return;
```
Comment (MartinNowak) [inline std.ddoc]: How about a comment with an example of what you're filtering out here?

File: std.ddoc (hunk 1 above)
Comment (ntrel) [inline std.ddoc]: Why?

File: std.ddoc (hunk 2 above)
Comment (ntrel) [inline std.ddoc]: The next line is `/**/ lastSymbolDefined = sym;`. Without the `/**/`, ddoc would end the current macro definition and start a new macro called `lastSymbolDefined`. Andrei used `if (true)` instead, but I thought an empty comment was slightly better. IMO ddoc should not allow macro definitions when there is indentation.

File: std.ddoc (hunk 1 above)
Comment (MartinNowak) [inline std.ddoc]: [javascript - Why is using "for...in" with array iteration such a bad idea? - Stack Overflow](http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea)

File: std.ddoc (hunk 2 above)
Comment (MartinNowak) [inline std.ddoc]: I think we should rather put it into a listanchors.js file.

File: std.ddoc (hunk 1 above)
Comment (ntrel) [inline std.ddoc]: Hmm, so it's slower, and iterates in order of element creation rather than index order.

File: std.ddoc (hunk 2 above)
Comment (ntrel) [inline std.ddoc]: Nice idea. I've now done that for my other branch #726, which includes this pull. I'll update this pull soon (ran out of time again today).

**Comment by MartinNowak** [comment]
Let's go on with this.

**Pull request merged**